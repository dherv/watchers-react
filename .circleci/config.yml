version: 2
jobs:
  build:
    working_directory: ~/tmp
    branches:
      only:
        - develop
        - feature/*
        - master
    docker:
      - image: circleci/node
    steps:
      - checkout
      - run: yarn install
      - run: yarn test
