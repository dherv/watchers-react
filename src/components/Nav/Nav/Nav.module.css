.list {
  padding: var(--padding);
  margin: var(--margin);
  box-shadow: var(--shadow);
  border-bottom: 1px solid #4db6ac;
}
