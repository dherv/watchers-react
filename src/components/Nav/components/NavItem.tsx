import React, { PureComponent } from "react";
import styles from "./NavItem.module.css";

interface IProps {
  item: string;
  onClick(item: string): void;
}
export default class NavItem extends PureComponent<IProps> {
  constructor(props: IProps) {
    super(props);
    this.handleClick = this.handleClick.bind(this);
  }
  handleClick() {
    this.props.onClick(this.props.item);
  }
  render() {
    return (
      <li className={styles.item}>
        <a className={styles.link} onClick={this.handleClick}>
          {this.props.item}
        </a>
      </li>
    );
  }
}
