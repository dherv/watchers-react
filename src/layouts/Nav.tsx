import React, { PureComponent, MouseEvent } from "react";
import styled from "styled-components";
import { Link } from "react-router-dom";

interface IProps {}

export default class MainNav extends PureComponent<IProps> {
  render() {
    return (
      <nav>
        <ul>
          <Link to="/movies">movies</Link>
          <Link to="/series">series</Link>
          <Link to="/likes">likes</Link>
          <Link to="/watchlist">watchlist</Link>
          <Link to="/account">account</Link>
        </ul>
      </nav>
    );
  }
}
