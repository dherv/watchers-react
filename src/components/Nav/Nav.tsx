import React, { SFC, MouseEvent, PureComponent } from "react";
import NavItem from "./components/NavItem";

const Nav: SFC<{
  items: string[];
  onClick(item: string): void;
}> = ({ items, onClick }) => (
  <nav>
    <ul>
      {items.map((item: string) => (
        <NavItem onClick={onClick} item={item} />
      ))}
    </ul>
  </nav>
);

export default Nav;
