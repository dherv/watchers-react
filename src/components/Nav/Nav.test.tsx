import { renderer, shallow, React } from "../../test.config";
import Nav from "./Nav";
import NavItem from "./components/NavItem";

describe("Nav", () => {
  const props = {
    items: [
      { text: "item1", value: "value1" },
      { text: "item2", value: "value2" }
    ],
    onClick: jest.fn()
  };
  const wrapper = shallow(<Nav {...props} />);
  describe("component", () => {
    it("should display one nav and one ul", () => {
      expect.assertions(2);
      expect(wrapper.find("ul").length).toBe(1);
      expect(wrapper.find("nav").length).toBe(1);
    });

    it("should display the right number of NavItem components", () => {
      expect.assertions(1);
      expect(wrapper.find(NavItem).length).toBe(2);
    });

    it("should call the prop onClick when a NavItem is clicked", () => {
      expect.assertions(1);
      const item = wrapper.find(NavItem).first();
      item.prop("onClick")("item1");
      expect(props.onClick).toHaveBeenCalledTimes(1);
    });
  });
  describe("snapshot", () => {
    test("should match", () => {
      const tree = renderer.create(<Nav {...props} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
