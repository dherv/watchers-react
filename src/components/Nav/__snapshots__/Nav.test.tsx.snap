// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav snapshot should match 1`] = `
<nav>
  <ul
    className="list"
  >
    <li
      className="item"
    >
      <div
        className="link"
        onClick={[Function]}
        role="menuitem"
      >
        item1
      </div>
    </li>
    <li
      className="item"
    >
      <div
        className="link"
        onClick={[Function]}
        role="menuitem"
      >
        item2
      </div>
    </li>
  </ul>
</nav>
`;
